pulsar-broker/src/main/java/org/apache/pulsar/broker/service/TopicListService.java

```diff
@@ -59,9 +65,12 @@ public List<String> getMatchingTopics() {
             return matchingTopics;
         }
 
+        /***
+         * @param topicName topic name which contains partition suffix.
+         */
         @Override
         public void accept(String topicName, NotificationType notificationType) {
-            if (topicsPattern.matcher(topicName).matches()) {
+            if (topicsPattern.matcher(TopicName.get(topicName).getPartitionedTopicName()).matches()) {
```

Comment:
Could we check both the partitioned topic name and the topic name with the partition suffix?

Reply:
@gaoran10 Ah, you means return `true` whether `{tenant}/{ns}/{tp}` is matches or `{tenant}/{ns}/{tp}-{partition}` is matches. Good point. We can submit a new PIP to do this(it is a behavior change)

This PR is in order to revet the behavior as the original behavior before [PIP-145](https://github.com/apache/pulsar/issues/14505)(it did a break change).